HotelManagement/src/core/Manager.java

```diff
@@ -59,15 +61,15 @@ public void setHotel(Hotel hotel) {
 	 * Books a room by room number.
 	 * @param roomNumber the number of the room to be booked
 	 */
-	public void bookRoom(int roomNumber) {
-		ArrayList<Room> rooms = this.hotel.getRooms();
+	public void bookRoom(int roomNumber, LocalDate fromDate, LocalDate toDate, int size, int numberOfBeds, String guestName, String guestId) {
+		List<Room> rooms = this.hotel.getRooms();
```

Comment:
You have only 1 hotel property in this scope, hence you don't need this.

Reply:
null checks for the dates

Reply:
Done